File: src/test/java/name/gyger/jmoney/util/DateUtilTests.java
package name.gyger.jmoney.util;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.assertThatThrownBy;

public class DateUtilTests {

    @Test
    public void testConstructor() {
        new DateUtil();
    }

    @Test
    public void testParsing() {
        assertThat(DateUtilKt.parse("2005-05-14")).isNotNull();
    }

    @Test
    public void testParseNull() {
        assertThat(DateUtilKt.parse(null)).isNull();
    }

    @Test
    public void testInvalidString() {
        assertThatThrownBy(() -> DateUtilKt.parse("INVALID STRING"))
                .isInstanceOf(IllegalArgumentException.class);
    }

}

File: src/test/kotlin/name/gyger/jmoney/util/DateUtilTests.kt
package name.gyger.jmoney.util

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.Assertions.assertThatThrownBy
import org.junit.Test

class DateUtilTests {

    @Test
    fun testConstructor() {
        DateUtil()
    }

    @Test
    fun testParsing() {
        assertThat(parse("2005-05-14")).isNotNull()
    }

    @Test
    fun testParseNull() {
        assertThat(parse(null)).isNull()
    }

    @Test
    fun testInvalidString() {
        assertThatThrownBy { parse("INVALID STRING") }.isInstanceOf(IllegalArgumentException::class.java)
    }

}
